sympy/polys/rootconditions.py

```diff
@@ -0,0 +1,264 @@
+from __future__ import annotations
+
+from sympy.polys.densebasic import dup
+from sympy.polys.domains.domain import Er, Domain
+from sympy.external.gmpy import MPQ
+
+from sympy.polys.domains import EXRAW, QQ, PolynomialRing, RationalField
+from sympy.polys.rings import PolyElement
+from sympy.polys.densebasic import dup_convert
+from sympy.polys.densetools import dup_clear_denoms
+from sympy.core.mul import prod
+
+from sympy.core.exprtools import factor_terms
+from sympy.simplify.simplify import signsimp
+from sympy.core.mul import Mul
+
+def dup_routh_hurwitz_stability(f: dup[Er], K: Domain[Er]) -> list[Er]:
+    """
+    Return the conditions for the polynomial to be stable (all roots with
+    negative real part).
+    The conditions, in general, are represented by a list of multivariate
+    polynomials which must be positive to ensure stability.
+
+    Note: This method assumes that the leading coefficient is non-zero.
+    In the opposite case, additional verification is required.
+
+    Depending on the domain, a different approach is used.
+    In non-numeric cases, the algorithm is modified to avoid divisions.
+
+    References
+    ==========
+    .. [1] G. Meinsma: Elementary proof of the Routh-Hurwitz test.
+           Systems & Control Letters, Volume 25, Issue 4, 1995, Pages 237-242,
+           https://courses.washington.edu/mengr471/resources/Routh_Hurwitz_Proof.pdf
+
+    """
+    if K.is_QQ:
+        return _routh_hurwitz_qq(f, K)
+
+    elif K.is_ZZ or K.is_RR:
+        pq = dup_convert(f, K, QQ)
+        conds = _routh_hurwitz_qq(pq, QQ)
+        return dup_convert(conds, QQ, K)
+
+    elif K.is_PolynomialRing:
+        return _routh_hurwitz_poly(f, K)
+
+    elif K.is_FractionField:
+        _, pp = dup_clear_denoms(f, K, convert=True)
+        conds = _routh_hurwitz_poly(pp, K)
+        return dup_convert(conds, K.get_ring(), K)
+
+    else:
+        return _routh_hurwitz_exraw(dup_convert(f, K, EXRAW))
+
+
+def _routh_hurwitz_qq(p: list[MPQ], K: RationalField) -> list[MPQ]:
```

Comment:
The functions here should use a consistent naming scheme. This should probably be called `dup_routh_hurwitz_qq` perhaps with a leading underscore.